File: frontend/src/types/todo.ts
export interface Todo {
  id: number
  user_id: number
  title: string
  content: string | null
  due_date: string | null
  is_completed: boolean
  created_at: string
  updated_at: string
}

export interface TodoListResponse {
  todos: Todo[]
}

export interface CreateTodoRequest {
  title: string
  content?: string
  due_date?: string
}

export interface UpdateTodoRequest {
  title: string
  content: string
  due_date?: string
  is_completed: boolean
}

export type FilterType = 'all' | 'incomplete' | 'completed' | 'overdue'

File: notification/domain/entity/todo.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Todo struct {
	ID          int            `gorm:"primaryKey;autoIncrement"`
	UserID      int
	Title       string
	Content     *string
	DueDate     *time.Time
	IsCompleted bool
	CreatedAt   time.Time      `gorm:"autoCreateTime"`
	UpdatedAt   time.Time      `gorm:"autoUpdateTime"`
	DeletedAt   gorm.DeletedAt
}
